src/Common/tests/gtest_helper_functions.h

```diff
@@ -0,0 +1,59 @@
+#pragma once
+
+#include "Poco/DOM/Document.h"
+#include "Poco/DOM/NodeList.h"
+#include "Poco/DOM/NamedNodeMap.h"
+
+
+inline char *getTempFileWithContents(const char *fileContents)
```

Comment:
Consider to use `DB::WriteBufferFromFile` with path starting with `/tmp/...`

Reply:
Do we need to take Windows builds into account or not needed? As on Windows the `/tmp/...` path wouldn't be possible.
Can also store files in some other place, but then the question is where? (preferably they get cleaned automatically if `remove` fails for whatever reason, and must be possible to consistently reference them from tests, which was failing with the lookup before)

Reply:
@nikitamikhaylov any idea how we can further improve this unit test? I believe external files are only used for functional / integration tests, nowhere else.
Initially I thought about using `YAMLParser::processNode` directly, but this is currently not exposed and I didn't want to change the interface for this class.
gtest apparently has something like `FRIEND_TEST` to expose internal functions just for testing. Not sure if we can / should use that instead?

Reply:
No, ClickHouse doesn't build for Windows.